### DISCUSSION TITLE
set x-message-ttl on upstream queue to "unlimited"

### DISCUSSION THREAD

**Discussion by ztdietmaier** [rabbitmq/messaging-topology-operator#346] [Q&A] [answered]
Hello, I'm trying to set up federation via the topology operator, and seem to somehow end up with an x-message-ttl of zero on the generated queue in the federation upstream. What I'd like to have is have no x-message-ttl on the queue.

I've tried the following values in the definition of the Federation resource:
  messageTTL: 315576000000 # results in x-message-ttl: 2043387392 -- the value seems to be limited
  messageTTL: null # results in x-message-ttl: 0, should ideally be not set
  not setting messagetTTL also results in x-message-ttl: 0

Is there a way to set the message TTL to "infinite" on the federation queue? Thank you!

**Comment by ChunyiLyu** [comment, marked as answer]
@ztdietmaier Sorry for the late reply. If you are still having this issue: topology operator only sets [federation upstream](https://www.rabbitmq.com/federation.html#tutorial), operator will never apply a message TTL configuration when you create a `federations.rabbitmq.com` resource. I suggest to double check all the policies you have in place in the rabbitmq cluster that you have this issue to see if there is a TTL policy that's matching the federation queue and see if you could update or remove the policy.

**Comment by ztdietmaier** [comment]
@ChunyiLyu no worries, this problem seems to be related to the federation plugin then -- but it became a moot point in the meantime, because we need to move away from exchange federation anyway. thanks!
